accounts/abi/argument.go

```diff
@@ -67,59 +67,55 @@ func (arguments Arguments) LengthNonIndexed() int {
 	return out
 }
 
+func (arguments Arguments) NonIndexed() Arguments {
```

Comment:
wouldn't it make sense to rewrite it as an iterator, since it's mostly called with `range` and the only time that you don't, you only need the first value? That would avoid parsing the entire array if you don't need to.

Reply:
@holiman Please add a method doc.

@gballet Usually a method will have only a handful of arguments. Creating and maintaining an iterator would be a much larger overhead than cycling through 5 elements of a slice.